### ISSUE TITLE
Feature Request: Relative dx (offset) tag parameter

### ISSUE THREAD

**Issue by arshiacont** [grame-cncm/guidolib#150]
This is a proposal for a useful addition to currently supported Tag Parameters in GuidoLib.

The idea is to extend `dx` to support a positive or negative _duration_ value (same syntax as note durations elsewhere). To make things simple: The duration, set by user, must be limited to the _span_ of the current measure in question.

**Why is this extension useful?** Some tags can only be positioned relatively in the score. Take the following screenshot. The Harmonies can currently be positioned in GuidoLib using (e.g.) `dx=-5hs`. But setting non-relative values will make the Guido Score dependent of Layout. This new feature will preserve GuidoLib ability to render dynamically and adapt to context (unique in its domain).

<img width="163" alt="image" src="https://user-images.githubusercontent.com/3298204/160984784-17ac9cb0-aa72-4411-ab17-388049e7527d.png">

**Potential Naïve Implementation?** GuidoLib engine can take the relative `dx` value, take its current voice position and simply use the rational offset with current measure length to convert that value to an absolute offset, at the last minute of rendering?!

**Comment by dfober** [comment]
Good idea! Implies to introduce a new unit: e.g. 'qn' (for quarter note). What might be complicated is that the value (in space) of a qn should always be available. I'll have a look.

**Comment by arshiacont** [comment]
Any chance to revive this? 😎

**Comment by arshiacont** [comment]
PS: Having this will enable using MusicXML `offset` tags which are relative to current voice position (based on current `divisions` attribute). It is widely used for _harmony_ and _directions_.

**Comment by dfober** [comment]
I don't know when I'll find the time to look into it, but yes, it's a good idea. It'll have to wait until mid-September at least.

**Comment by arshiacont** [comment]
PR #199 addresses this feature. There is a small unit test `g-issue150.gmn` in regression test folder that simulates the behavior.

**Comment by dfober** [comment]
I've made some progress on the problem. You can see it on the [relative_location](https://github.com/grame-cncm/guidolib/tree/relative_location) branch. 
For now it only works for the \intens tag and I think there is no general solution given that getDx() is on the AR side. However, I think we can generalize a little more.
A few questions:

- Which tags should support relative DX?
- Currently, relative dx is expressed using a unit called "rl" (for relative location). The unit of the given value is the quarter note. But I wonder if it wouldn't be more useful to give a fraction of the current duration, i.e., "0.5rl" would represent a quarter note when applied to a half note, or an eighth note for a quarter note. How do MusicXML express that?
